src/osx/textctrl_osx.cpp

```diff
@@ -107,7 +107,7 @@ bool wxTextCtrl::Create( wxWindow *parent,
         SetEditable( false ) ;
 
     SetCursor( wxCursor( wxCURSOR_IBEAM ) ) ;
-
+    m_maxlen = UINT_MAX;
```

Comment:
I'd initialize it to 0 to indicate that it's not set and do it in the declaration.